lint_test/avoid_unnecessary_type_casts_test.dart

```diff
@@ -0,0 +1,18 @@
+// ignore_for_file: prefer_const_declarations
+// ignore_for_file: unnecessary_nullable_for_final_variable_declarations
+// ignore_for_file: unnecessary_cast
+// ignore_for_file: unused_local_variable
+
+/// Check the `avoid-unnecessary-type-casts` rule
+
+void fun() {
+  final testList = [1.0, 2.0, 3.0];
+
+  // to check quick-fix => testList
+  // expect_lint: avoid-unnecessary-type-casts
+  final result = testList as List<double>;
+
+  final double? nullableD = 2.0;
+  // casting `Type? is Type` is allowed
+  final castedD = nullableD as double;
+}
```

Comment:
Let's try to add more test cases.

For one, Dart Code Checker often stumbled on Map handling.

```
const m = {
  'a' : 'b'
 }

final mVal = m['a'] as String?;
```

I'd also check how this thing works with closures, passing arguments into a function etc.